strawberryfields/apps/graph/sample.py

```diff
@@ -0,0 +1,139 @@
+) -> list:
+    """Functionality for sampling subgraphs.
+
+    The optional ``sample_options`` argument can be used to specify the type of sampling. It
+    should be a dict that contains any of the following:
+
+    .. glossary::
+
+        key: ``"distribution"``, value: *str*
+            Subgraphs can be sampled according to the following distributions:
+
+            - ``"gbs"``: for generating subgraphs according to the Gaussian boson sampling
+              distribution. In this distribution, subgraphs are sampled with a variable size (
+              default).
+            - ``"uniform"``: for generating subgraphs uniformly at random. When using this
+              distribution, subgraphs are of a fixed size. Note that ``backend_options`` are not
+              used and that remote sampling is unavailable due to the simplicity of the
+              distribution.
+
+        key: ``"postselect_ratio"``, value: *float*
+            Ratio of ``nodes`` used to determine the minimum size of subgraph sampled; defaults
+            to 0.75.
+
+    Args:
+        graph (nx.Graph): the input graph
+        nodes (int): the mean size of subgraph samples
+        samples (int): number of samples; defaults to 1
+        sample_options (dict[str, Any]): dictionary specifying options used by ``sample_subgraphs``;
+            defaults to :const:`SAMPLE_DEFAULTS`
+        backend_options (dict[str, Any]): dictionary specifying options used by backends during
+            sampling; defaults to ``None``
+
+    Returns:
+        list[list[int]]: a list of length ``samples`` whose elements are subgraphs given by a
+        list of nodes
+    """
+    sample_options = {**SAMPLE_DEFAULTS, **(sample_options or {})}
+
+    distribution = sample_options["distribution"]
+
+    if distribution == "uniform":
+        s = sample.uniform_sampler(
+            modes=graph.order(), sampled_modes=nodes, samples=samples
+        )
+    elif distribution == "gbs":
+        postselect = int(sample_options["postselect_ratio"] * nodes)
+        backend_options = {**(backend_options or {}), "postselect": postselect}
+
+        s = sample.quantum_sampler(
+            A=nx.to_numpy_array(graph),
+            n_mean=nodes,
+            samples=samples,
+            backend_options=backend_options,
+        )
+    else:
+        raise ValueError("Invalid distribution selected")
+
+    return to_subgraphs(graph, s)
+
+
+def to_subgraphs(graph: nx.Graph, samples: list) -> list:
+    """Converts a list of samples to a list of subgraphs.
+
+    Given a list of samples, with each sample of ``len(nodes)`` being a list of zeros and ones,
+    this function returns a list of subgraphs selected by, for each sample, picking the nodes
+    corresponding to ones and creating the induced subgraph. The subgraph is specified as a list
+    of selected nodes. For example, given an input 6-node graph, a sample :math:`[0, 1, 1, 0, 0,
+    1]` is converted to a subgraph :math:`[1, 2, 5]`.
```

Comment:
Something to keep in mind for future PRs (I think @ixfoduap mentioned this as some point?) but it might be nice to have (short) code examples showing how to use the function within the docstring. This is something we can also work on in Strawberry Fields and PennyLane.